godot-core/src/builtin/aabb.rs

```diff
@@ -43,32 +43,346 @@ impl Aabb {
+        } else {
+            -half_extents.z
+        };
+
+        Vector3::new(x, y, z) + ofs
+    }
+
+    /// Returns `true` if the AABB overlaps with `b` (i.e. they have at least one point in common).
+    ///
+    /// If `include_borders` is `true`, they will also be considered overlapping if their borders touch, even without intersection.
+    ///
+    /// _Godot equivalent: `AABB.intersects(AABB b, bool include_borders)`_
+    #[inline]
+    pub fn intersects(&self, &b: &Aabb, include_borders: bool) -> bool {
+        let end = self.end();
+        let end_b = b.end();
+
+        if include_borders {
+            self.position.x <= end_b.x
+                && end.x >= b.position.x
+                && self.position.y <= end_b.y
+                && end.y >= b.position.y
+                && self.position.z <= end_b.z
+        } else {
+            self.position.x < end_b.x
+                && end.x > b.position.x
+                && self.position.y < end_b.y
+                && end.y > b.position.y
+                && self.position.z < end_b.z
+                && end.z > b.position.z
+        }
+    }
+
+    /// Returns `true` if the AABB is on both sides of a plane.
+    ///
+    /// _Godot equivalent: `AABB.intersects_plane(Plane plane)`_
+    pub fn intersects_plane(&self, plane: Plane) -> bool {
+        let points = [
+            self.position,
+            self.position + Vector3::new(0.0, 0.0, self.size.z),
+            self.position + Vector3::new(0.0, self.size.y, 0.0),
+            self.position + Vector3::new(self.size.x, 0.0, 0.0),
+            self.position + Vector3::new(self.size.x, self.size.y, 0.0),
+            self.position + Vector3::new(self.size.x, 0.0, self.size.z),
+            self.position + Vector3::new(0.0, self.size.y, self.size.z),
+            self.position + self.size,
+        ];
+
+        let mut over = false;
+        let mut under = false;
+
+        for point in points {
+            // TODO: Replace when Plane.distance_to() is implemented
+            println!("{}", point);
+            let dist_to = plane.normal.dot(point - plane.d * plane.normal);
+            if dist_to > 0.0 {
+                over = true
+            } else {
+                under = true
+            }
+        }
+
+        over && under
+    }
+
+    /// Returns `true` if the given ray intersects with this AABB. Ray length is infinite.
+    ///
+    /// _Godot equivalent: `AABB.intersects_ray(Vector3 from, Vector3 dir)`_
+    pub fn intersects_ray(&self, from: Vector3, dir: Vector3) -> bool {
+        self.assert_nonnegative();
+
+        let tmin = (self.position - from) / dir;
+        let tmax = (self.end() - from) / dir;
+
+        let t1 = tmin.coord_min(tmax);
+        let t2 = tmin.coord_max(tmax);
+
+        let tnear = t1.x.max(t1.y).max(t1.z);
+        let tfar = t2.y.min(t2.x).min(t2.z);
+
+        tnear <= tfar
+    }
+
+    pub fn intersects_segment(&self, from: Vector3, to: Vector3) -> bool {
```

Comment:
Missing documentation. Would be good to highlight the difference to `intersects_ray`.